superset/security/api.py

```diff
@@ -35,7 +35,7 @@ class UserSchema(Schema):
 
 
 class ResourceSchema(Schema):
-    type = fields.String(required=True)
+    type = fields.String(required=True)  # todo figure out how to make this an enum
```

Comment:
You can use a [validator](https://marshmallow.readthedocs.io/en/stable/marshmallow.validate.html) here:
```python
class ResourceSchema(Schema):
    type = Fields.String(
        required=True,
        validate=validate.OneOf(choices=("option_1", "option_2", "option_3")),
    )
```

Reply:
Even better, the `marshmallow_enum` package adds an `EnumField` https://github.com/apache/superset/pull/17836/files#diff-ddc89915e8af096a98449594cef72e9d33b26bfd37de6b549fd677000a24ee9bR52